sql/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java

```diff
@@ -73,13 +73,20 @@
     @Nullable
     private Streamer[] insertValuesStreamer;
 
+    /**
```

Comment:
`ShardUpsertRequest` contains already the information for a lot of use cases. It would make sense to split this eventually this as @mfussenegger already suggested.

Reply:
I would probably tackle this later.